docs/concepts.rst

```diff
@@ -220,6 +220,7 @@ execution_date
 The ``execution_date`` is the *logical* date and time which the DAG Run, and its task instances, are running for.
 
 This allows task instances to process data for the desired *logical* date & time.
+//TODO: rewrite this sentence to something understandable! The idea is too tricky!!! We are busy doing (did not ketch what we are doing). That is why the dag which supposed to be ran 3 monthes ago actually ran now?? Or what?? What all this words meen???
```

Comment:
@aboronnikov would you like to propose more clear explanation? This is an open source project and we are happy when our users help us improve the code and documentation 💪

Reply:
I would propose some better explanation in case if I understand all this stuff. But I did not understand it at all!!!!

Reply:
It would be great if you can point which part of this sentence you do not understand. I believe that we can together work out a better documentation to avoid such frustration in future. This however, requires you to explain us what is not understandable. Otherwise we face the risk that we will again write something that others cannot understand.

Reply:
On the side note - there are better places to get you questions answered than question in comments in code:
https://github.com/apache/airflow/issues/new/choose